src/test/java/org/mozilla/jss/tests/CapabilitiesList.java

```diff
@@ -0,0 +1,260 @@
+                    new File(System.getProperty("user.dir").concat("/listings/verbose"));
+                    dir4verboseListings.mkdir();
+
+                File dir4briefListings = 
+                    new File(System.getProperty("user.dir").concat("/listings/brief"));
+                dir4briefListings.mkdir();
+
+                return true;
+
+            } catch (Exception e) {
+                logger.info("Exception caught in createOutputDirs: " + e.getMessage(), e);
+                logger.info("Keep going");
+                return false;
+            }
+        }
+
+        /* List providers capabilities using the brief listing method which adds
+         * results for each provider listed to the listings/brief directory
+         */
+        public void listBrief(Provider[] ps) throws Exception {
+            try {
+                for (int i = 0; i < ps.length; i++) {
+                    String fileName = briefFileBase + ps[i].getName() + ".txt";
+                    FileWriter fw = new FileWriter(new File(fileName));
+                    for (Enumeration e = ps[i].keys(); e.hasMoreElements();) {
+                        fw.write(String.format("\t %s", e.nextElement()));
+                        fw.write(System.lineSeparator());
+                    }
+                    fw.close();
+                    File resultsFile = new File(fileName);
+                    assert(resultsFile.exists());
+                }
+            } catch (Exception e) {
+                logger.info("Exception caught in listBrief: " + e.getMessage(), e);
+                logger.info("Keep going");
+            }
+        }
+
+        /* List providers capabilities using the verbose listing method which
+         * adds results for each provider listed to the listings/verbose directory
+         */
+        public void listVerbose(Provider[] ps) throws Exception {
+            try {
+                for (int i = 0; i < ps.length; i++) {
+                    String fileName = verboseFileBase + ps[i].getName() + ".txt";
+                    FileWriter fw = new FileWriter(new File(fileName));
+                    listCapabilities(fw, ps[i]);
+                    fw.close();
+                    File resultsFile = new File(fileName);
+                    assert(resultsFile.exists());
+                }
+            } catch (Exception e) {
+                logger.info("Exception caught in listVerbose: " + e.getMessage(), e);
+                logger.info("Keep going");
+            }
+        }
+    }
+
+    public static void main(String[] args) {
+        try {
+            Capabilities lister = new Capabilities();
+            if (!lister.createOutputDirs()) return;
+            lister.addJssProvider();
+            Provider ps[] = Security.getProviders();
+            lister.listBrief(ps);
+            lister.listVerbose(ps);
+        } catch (Exception e) {
+            Capabilities.logger.info("Exception caught in main: " + e.getMessage(), e);
+        }
```

Comment:
I'd suggest removing this `try-catch` block too so if something fails the program will return a non-zero code.